lib/LTP/utils.pm

```diff
@@ -432,4 +421,21 @@ sub install_from_repo {
     }
 }
 
+sub prepare_whitelist_environment {
+    my $environment = {
+        product => get_var('DISTRI') . ':' . get_var('VERSION'),
+        revision => get_var('BUILD'),
+        flavor => get_var('FLAVOR'),
+        arch => get_var('ARCH'),
+        backend => get_var('BACKEND'),
+        kernel => script_output('uname -r'),
```

Comment:
You should not use `script_output()` or any `script_run()` calls here. This function may be called e.g. in public cloud where the command needs to be wrapped in an ssh command.

Reply:
sorry but I don't get the previous comment then. You meant to populate `kernel` item after we called `prepare_whitelist_environment()` ?

Reply:
Yes. `$environment->{kernel}` needs to be set in `kselftest::run()`.

Reply:
Ok, moved that assignment into `ebpf.pm`. Please take a look if we need more modifications